lib/ansible/modules/cloud/vmware/vmware_host_tcpip_stacks.py

```diff
@@ -0,0 +1,473 @@
+                                congestion_algorithm_previous
+                        changed_instances_list.append(netstack_spec)
+                # Check VXLAN TCP/IP stack
+                if instance.key == 'vxlan':
+                    results['result'][host.name]['vxlan_vmkernel_gateway'] = vxlan_vmkernel_gateway
+                    results['result'][host.name]['vxlan_vmkernel_num_connections'] = vxlan_num_connections
+                    results['result'][host.name]['vxlan_congestion_algorithm'] = vxlan_congestion_algorithm
+                    (netstack_spec,
+                     changed_instance,
+                     vmkernel_gateway_previous,
+                     num_connections_previous,
+                     congestion_algorithm_previous) = self.create_netstack_spec(
+                         instance,
+                         'vxlan',
+                         vxlan_vmkernel_gateway,
+                         vxlan_num_connections,
+                         vxlan_congestion_algorithm)
+                    if changed_instance:
+                        changed = True
+                        changed_list.append("VXLAN TCP/IP stack")
+                        # vmkernel_gateway_previous can be Null
+                        if vxlan_vmkernel_gateway:
+                            results['result'][host.name]['vxlan_vmkernel_gateway_previous'] = vmkernel_gateway_previous
+                        if num_connections_previous:
+                            results['result'][host.name]['vxlan_num_connections_previous'] = num_connections_previous
+                        if congestion_algorithm_previous:
+                            results['result'][host.name]['vxlan_congestion_algorithm_previous'] = \
+                                congestion_algorithm_previous
+                        changed_instances_list.append(netstack_spec)
+
+            if changed:
+                config = vim.host.NetworkConfig()
+                config.netStackSpec = changed_instances_list
+                if self.module.check_mode:
+                    changed_suffix = ' would be changed'
+                else:
+                    changed_suffix = ' changed'
+                if len(changed_list) > 2:
+                    message = ', '.join(changed_list[:-1]) + ', and ' + str(changed_list[-1])
+                elif len(changed_list) == 2:
+                    message = ' and '.join(changed_list)
+                elif len(changed_list) == 1:
+                    message = changed_list[0]
+                message += changed_suffix
+                results['result'][host.name]['changed'] = True
+                host_network_system = host.configManager.networkSystem
+                if not self.module.check_mode:
+                    try:
+                        host_network_system.UpdateNetworkConfig(config, 'modify')
+                    except vim.fault.AlreadyExists:
```

Comment:
How do we test the failure conditions?  Can you simulate the vim.fault responses with the simulator?

Reply:
No we can not simulate faults on simulator. Real ESXi is source of this error replication.

Reply:
We need to start adding unit testing to these modules then - nobody can prove how they behave when things go wrong, now or in the future.  Not saying we need to for this specific PR, just in general.  The simulator is clearly lacking in fault injection techniques and insufficient to qualify all parts of the vmware modules.

Reply:
Nobody has time for this anyway...

Reply:
It is a valid concern of @jeking3 ... Our testing has been spotty and we've had bugs and issues because of that. This is the hard part of developing against external systems, but having something that can cover as much as possible of the module is important to make sure everything behaves as expected.

Not something to address/discuss here, but it is something we need to remain aware off when developing and in our meetings